modevo-consistency/src/test/java/test4giis/consistency/TestCheckConsistency.java

```diff
@@ -43,103 +45,140 @@ public static void setUp(){ //cleans the entire DB
+			String query = tableQuery.get(tableName);
+			new Oracle().sqlQueryMigrate(query, connectionSql, connection, preparedStatement);
+		}
+		try {
+			connectionSql.close();
+		} catch (SQLException e) {
+			throw new ScriptException(e);
 		}
-		connectionSQL.close();
-
 	}
     /**
-     * Method to call the oracle. Right now it includes hardcodded the query for the only experimental subject tested. 
-     * In a following version this will be generalized and obtained from an external source depending on the experimental subject.
-     * @param tableQuery 
-     * @param keyspace 
+     * Calls the transform and script module for the migration determined by MoDEvo
      */
-    private void compareCassandraSQL(Map<String, String> tableQuery, String keyspace) throws Exception{
-		MySQLAccess mysql = new MySQLAccess();
-		Connection con = mysql.connect(keyspace);
-		Oracle oc = new Oracle();
-		boolean result = oc.oracleCompare(tableQuery, name.getMethodName(), PROPERTIES, con);
-		con.close();
-		assertTrue (result);
+    private void testScript (String testName,  CassandraConnection c) {
+		ModelObjects m = new TestUtils().executeTransformationsAndCompareOutput (testName);
+		new MainScript().createScriptAndText(m, c, testName);
 	}
+    
     /**
-     * Call the transform and script module for the migration determined by MoDEvo
+     * Compares the data stored in the Cassandra database and the SQL database by comparing
+     * each table of the Cassandra database with its projection from the SQL database.
      */
-    private void testScript (String nameTest,  CassandraConnection c) {
-		ModelObjects m = new TestUtils().executeTransformationsAndCompareOutput (nameTest);
-		new MainScript().createScriptAndText(m, c, nameTest);
+    private void compareCassandraSql(Map<String, String> tableQuery, String keyspace) {
+		OracleConnection mysql = new OracleConnection();
+		Connection con = mysql.connect(keyspace);
+		String path = "dat/out/" + keyspace + "/";
+		OracleCsv csv = new OracleCsv();
+		Iterator<Entry<String, String>> iteratorTableQuery = tableQuery.entrySet().iterator();
+		while (iteratorTableQuery.hasNext()) {
+			Entry<String, String> tableQueryIteration = iteratorTableQuery.next();
+			String table= tableQueryIteration.getKey();
+			String query = tableQueryIteration.getValue();
+			String sqlPath = path + table + "SQL.csv";
+			String cassandraPath = path + table + "CQL.csv";
+			new Oracle().sqlQuery(query, con, sqlPath); //Creates a CSV with the projection data
+			csv.csvCassandra(table, name.getMethodName(), PROPERTIES, cassandraPath);
+			try {
+				//Retrieves the data from the csv files to the variables sql and cassandra
+				String sql = Files.readString(Paths.get(sqlPath)); 
+				String cassandra = Files.readString(Paths.get(cassandraPath));
+				VisualAssert va = new VisualAssert();
+				va.assertEquals(sql, cassandra); //compares the data stored in both files
```

Comment:
@javiertuya What do you mean by letting Junit to manage the exceptions? Use **throws** instead of managing the exception? In that case I would need to use throws for all the previous methods, would that be ok?

Reply:
@suarezgpablo Yes, as JUnit already captures the java managed exceptions